src/Components/DatePicker/Component.purs

```diff
@@ -0,0 +1,428 @@
+          -- a message about its selection.
+          H.liftEffect $ log ("Selected! Choice was " <> show date)
+          H.modify_ _ { selection = Just date }
+          _ <- H.query unit $ Select.setVisibility Select.Off
+          H.raise $ SelectionChanged $ Just date
+          eval $ Synchronize a
+
+        Select.Searched text -> do
+          H.modify_ _ { search = text }
+          -- we don't actually want to match on search, we want to wait
+          -- until they hit ENTER and then we'll try to match their search
+          pure a
+
+        Select.VisibilityChanged visibility -> do
+          H.raise $ VisibilityChanged visibility
+          pure a
+
+      -- We ought to be able to navigate months in the date picker
+      ToggleMonth dir a -> a <$ do
+        st <- H.get
+        let y = fst st.targetDate
+            m = snd st.targetDate
+            newDate = case dir of
+               Next -> ODT.nextMonth (canonicalDate y m bottom)
+               Prev -> ODT.prevMonth (canonicalDate y m bottom)
+        H.modify_ _ { targetDate = Tuple (year newDate) (month newDate) }
+        eval $ Synchronize a
+
+      -- We ought to be able to navigate years in the date picker
+      ToggleYear dir a -> a <$ do
+        st <- H.get
+        let y = fst st.targetDate
+            m = snd st.targetDate
+            newDate = case dir of
+               Next -> ODT.nextYear (canonicalDate y m bottom)
+               Prev -> ODT.prevYear (canonicalDate y m bottom)
+        H.modify_ _ { targetDate = Tuple (year newDate) (month newDate) }
+        eval $ Synchronize a
+
+      -- We can always set the calendar to the current date and time for the user's
+      -- convenience. Or just the month.
+      Initialize a -> do
+        { selection } <- H.get
+        d <- H.liftEffect nowDate
+        let d' = fromMaybe d selection
+        H.modify_ _ { targetDate = Tuple (year d') (month d') }
+        eval $ Synchronize a
+
+      TriggerFocus a -> a <$ H.query unit Select.triggerFocus
+
+      Synchronize a -> do
+        { targetDate: Tuple y m, selection } <- H.get
+        _ <- H.query unit
+          $ Select.replaceItems
+          $ generateCalendarRows selection y m
+        _ <- case selection of
+          Just date -> H.modify_ _ { search = ODT.formatDate date }
+          otherwise -> pure unit
+        pure a
+
+      GetSelection reply -> do
+        { selection } <- H.get
+        pure $ reply selection
+
+      SetSelection selection a -> do
+        st <- H.get
+        let targetDate = maybe st.targetDate (\d -> Tuple (year d) (month d)) selection
+        H.modify_ _ { selection = selection, targetDate = targetDate }
+        eval $ Synchronize a
+
+      Key ev a -> do
+        _ <- H.liftEffect $ log $ "key press: " <> show (KE.code ev)
+        _ <- H.query unit $ Select.setVisibility Select.On
+        let preventIt = H.liftEffect $ preventDefault $ KE.toEvent ev
+        case KE.code ev of
```

Comment:
Something fancy might be possible here to make key navigation doable. For example:

- Arrow keys navigate
- Escape key closes the menu (this one seems like it definitely ought to be there)
- Enter key performs a search
- Enter key, if this search is the same as the last search, selects the current index

@foresttoney?